indico_jacow/controllers.py

```diff
@@ -270,3 +277,101 @@ def _process(self, file):
             'identifiers': identifiers,
             'unknown_emails': list(unknown_emails)
         })
+
+
+class RHBrevoAPI(RH):
+    def __init__(self):
+        from indico_jacow.plugin import JACOWPlugin
+
+        super().__init__()
+        configuration_brevo = brevo_python.Configuration()
+        configuration_brevo.api_key['api-key'] = JACOWPlugin.settings.get('brevo_api_key')
+        self.api_instance = brevo_python.ContactsApi(brevo_python.ApiClient(configuration_brevo))
+
+
+class RHMailingLists(RHUserBase, RHBrevoAPI):
+    def _process(self):        
+        valid_contact_ids = set()
+        emails = self.user.all_emails
+
+        for email in emails:
+            try:
+                contact_info = self.api_instance.get_contact_info(email)
+                contact_data = contact_info.to_dict()
+                if 'list_ids' in contact_data:
+                    valid_contact_ids.update(contact_data['list_ids'])
+            except ApiException as e:
+                print(f'Warning: Contact {email} not found in Brevo. Skipping...')
+                return {'error': e.reason}, e.status
+
+        # Get all Brevo lists
+        try:
+            api_response_2 = self.api_instance.get_lists()
+            response_2_dict = api_response_2.to_dict()
+        except ApiException as e:
+            print(f'Exception when calling ContactsApi->get_lists: {e.reason}')
+            return {'error': e.reason}, e.status
+
+        # Mark subscribed lists
+        for lst in response_2_dict.get('lists', []):
+            if lst['id'] in valid_contact_ids:
+                lst['subscribed'] = True
+            else:
+                lst['subscribed'] = False
+
+        # Convert to JSON and return
+        mailing_lists = json.dumps(response_2_dict)
+        return WPUserMailingLists.render_template('mailing_lists.html', 'mailing_lists', user=self.user,
+                                                  mailing_lists=mailing_lists)
+
+
+class RHMailingListSubscribe(RHUserBase, RHBrevoAPI):
+    def _process(self):
+        email = self.user.email
+        print(email)
+        try:
+            list_id = int(request.args.get('list_id'))
```

Comment:
Use `@use_kwargs` for this.

Also, are you using a single list id or an array of list ids in the end (backend and frontend are doing different things atm). If it's a single one, I'd put it in the URL and use PUT/DELETE verbs to subscribe/unsubscribe. Otherwise sending it as POSTed data to your `.../subscribe` and `.../unsubscribe` endpoints is fine.